File: CSharp/OpenTraceability.GDST/Events/GDSTAggregationEvent.cs
using OpenTraceability.Models.Events;
using OpenTraceability.Models.Identifiers;
using OpenTraceability.Utility.Attributes;

namespace OpenTraceability.GDST.Events
{
    public class GDSTAggregationEvent : AggregationEvent<GDSTILMD>, IGDSTProductOwnerEvent
    {
        [OpenTraceability(Constants.GDST_NAMESPACE, "productOwner")]
        [OpenTraceabilityJson("gdst:productOwner")]
        public PGLN? ProductOwner { get; set; }

        public GDSTAggregationEvent()
        {
            BusinessStep = new Uri("urn:epcglobal:cbv:bizstep:packing");
            Action = EventAction.ADD;
            Disposition = new Uri("active", UriKind.RelativeOrAbsolute);
        }
    }
}

File: Java/opentraceability/src/main/java/opentraceability/gdst/events/GDSTAggregationEvent.java
package opentraceability.gdst.events;

import opentraceability.Constants;
import opentraceability.models.events.AggregationEventBase;
import opentraceability.models.events.EventAction;
import opentraceability.models.events.ObjectEventBase;
import opentraceability.models.events.TransformationEventBase;
import opentraceability.models.identifiers.PGLN;
import opentraceability.utility.attributes.*;

import java.net.URI;

public class GDSTAggregationEvent extends AggregationEventBase<GDSTILMD> implements IGDSTEvent  {
    @OpenTraceabilityAttribute(ns=Constants.GDST_NAMESPACE, name="productOwner")
    @OpenTraceabilityJsonAttribute(name="gdst:productOwner")
    private PGLN productOwner;

    @Override
    public PGLN getProductOwner() {
        return this.productOwner;
    }

    // Add constructor, getters, and setters here
    public GDSTAggregationEvent()
    {
        this.businessStep = URI.create("urn:epcglobal:cbv:bizstep:packing");
        this.action = EventAction.ADD;
    }
}

